src/contrib/cluster/Akka.Cluster.Sharding/PersistentShardCoordinator.cs

```diff
@@ -627,22 +698,45 @@ private void HandleRegister(Register message)
                     _gracefullShutdownInProgress = _gracefullShutdownInProgress.Remove(region);
                     Update(new ShardRegionRegistered(region), e =>
                     {
-                        var isFirstRegion = _currentState.Regions.Count == 0;
                         _currentState = _currentState.Updated(e);
                         context.Watch(region);
                         region.Tell(new RegisterAck(self));
 
-                        if (isFirstRegion) AllocateShardHomes();
+                        AllocateShardHomesForRememberEntities();
                     });
                 }
             }
             else Log.Debug("ShardRegion [{0}] was not registered since the coordinator currently does not know about a node of that region", region);
         }
 
+        private void HandleGetClusterShardingStats(GetClusterShardingStats message)
+        {
+            var sender = Sender;
+            Task.WhenAll(
+                _aliveRegions.Select(regionActor => regionActor.Ask<ShardRegionStats>(GetShardRegionStats.Instance, message.Timeout).ContinueWith(r => Tuple.Create(regionActor, r.Result), TaskContinuationOptions.ExecuteSynchronously))
+                ).ContinueWith(allRegionStats =>
+                {
+                    if (allRegionStats.IsFaulted || allRegionStats.IsCanceled)
+                    {
+                        return new ClusterShardingStats(ImmutableDictionary<Address, ShardRegionStats>.Empty);
```

Comment:
```
.recover {
    case x: AskTimeoutException ⇒ ShardRegion.ClusterShardingStats(Map.empty)
}.pipeTo(sender())
```
In case of different exception, doesn't it pipe the exception with the PipeTo?
I'm not scala friend so not sure :)